### PR TITLE
Functors: Task_1 Sarandi Ihor

### PR DESCRIPTION
**Pull request by IgorSarandi**
Signed-off-by: Sarandi Ihor <isarandi@luxoft.com>

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()() const;
+
+    private:
+    mutable int8_t counter = static_cast<int8_t>(BeerBrand::None);
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:15]: _prefer size_t_
why mutable?

File: functors task (1)/student/src/impl.cpp [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+BeerBrand BeerOrganizer::operator()() const
+{
+    if((counter + 1) >= static_cast<int8_t>(BeerBrand::Max))
+    {
+        counter = static_cast<int8_t>(BeerBrand::None);
+    }
+    return static_cast<BeerBrand>(++counter);
+}
+
+std::map <std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail> MixingPolicy::bar =
```
Comment (smay1613) [inline functors task (1)/student/src/impl.cpp:12]: Mapping is the best solution here, so you're right.
It is more flexible switch logic.

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (IgorSarandi) [reply functors task (1)/student/include/impl.h:15]: mutable because I declare BeerBrand operator()() as const method

Comment (smay1613) [reply functors task (1)/student/include/impl.h:15]: I think mutable is not a good solution here.
In fact, you modify the internal state, so it shouldn't be const.

Comment (IgorSarandi) [reply functors task (1)/student/include/impl.h]: Done. Its make sense, thank.

**Pull request closed without merging**